ARCHITECTURE.md

```diff
@@ -60,12 +52,43 @@ Rojo has many data structures and their purpose might not be immediately clear a
 
 To learn more, read about [`memofs` architecture](crates/memofs/ARCHITECTURE.md).
 
+### LiveServer
+
+LiveServer underlies the [`serve` command](#the-serve-command) and provides the web server which clients (such as the plugin) can use to interface with [`ServeSession`](#servesession).
+
+There's two parts to the API, the UI and the API clients use.
+
+The UI provides information about the current [`DOM`](#rojotree) has, including metadata. It also shows the project name, up-time, and version its Rojo is on.
+
+The API provides a simple JSON protocol to interact with and receive changes from the underlying [`ServeSession`](#servesession). Checkout the [`api.rs` file under the web module](src/web/api.rs) to learn more.
+
 ### ServeSession
 
+The linchpin of Rojo. It contains all of the required components to serve a given project file.
+
+Generally, to serve means:
+
+- Rojo maintains a DOM and exposes it to consumers;
+- Rojo is able to accept events to cause changes to the DOM;
+- Rojo is able to emit changes to the DOM to consumer.
+
+It depends on:
+
+- `RojoTree` to represent the DOM;
+- `Project` to represent your root project file (e.g. `default.project.json`);
+- [`Vfs`](#vfs) to provide a filesystem and emit events on changes;
+- [`ChangeProcessor`](#changeprocessor) to process filesystem events from `Vfs` and consequently update the DOM through the [snapshotting system](#the-snapshotting-system);
+
+It also provides an API for the higher level components so it can be used with the outside world.
+
+- There is a `MessageQueue` of changes applied to the DOM.
+- There is a channel to send changes to the `ServeSession` and update the DOM.
+- And a `SessionId` to uniquely identify the `ServeSession`.
+
+All of the public interfaces via CLI of Rojo are implemented using `ServeSession`.
```

Comment:
```suggestion
The primary interface for a `ServeSession` is the web server used by the [`serve` command](#servecommand). Additionally, several of Rojo's commands are implemented using `ServeSession`.
```

We should avoid getting too technical here. None of Rojo's existing commands actually interact directly with the `MessageQueue`, it's mostly an implementation detail.

Reply:
I see where you're coming from about getting too technical. I like the better summary you put it in. However, I think it would be weird omitting the other half of the `ServeSession` API. Especially because there wouldn't balance the first part of the internals and that most of Rojo's command do use to enable watching.

Reply:
I suppose we could mention that it has a `MessageQueue`, though I'm inclined to just explain what that is rather than just saying "there is a `MessageQueue`"

Reply:
Alright, so I think I applied what was suggested. What's your thoughts on it?